src/App.tsx

```diff
@@ -1,7 +1,39 @@
 import React from 'react';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <p className="App__message">The last pressed key is [Enter]</p>
-  </div>
-);
+type State = {
+  currentButton: string | null;
+};
+
+export class App extends React.Component<{}, State> {
+  state: State = {
+    currentButton: null,
+  };
+
+  UNSAFE_componentWillMount(): void {
+    document.removeEventListener('keyup', this.handleCurrentBatton);
+  }
```

Comment:
- you should use componentWillUnmount to delete the event listener (+ unsafe won't be needed)
- it's button, not batton